internal/pkg/api/handleAck.go

```diff
@@ -338,6 +338,15 @@ func (ack *AckT) handlePolicyChange(ctx context.Context, zlog zerolog.Logger, ag
 		return nil
 	}
 
+	err := ack.updateAgentDoc(ctx, zlog,
+		agent.Id,
+		currRev, currCoord,
+		agent.PolicyID)
```

Comment:
I'm not saying you need to change, but I'm not sure if doing it here could be a problem.

Reply:
another thing, do you think you could add a test to endure:
 - the document is updated
 - `updateAgentDoc` is called in the right moment

Reply:
I think it's better to update the agent doc to know which policy it's running even if we fail to invalidate retired API keys, what do you think?

Reply:
I think we might run into trouble. The agent document would state policy `x+1`, but the API keys would have permission for policy `x`. I cannot access the impact of it, but I don't think it's a good thing regardless of the impact.

Reply:
Let me ask if someone in my team can foresee any issue with either option

Reply:
from what I understand the code here after is invalidating old API keys, I think the worse case scenario will be: The agent document would state policy x+1, with old API keys still valid.

Reply:
I didn't get any extra input from the team. However I'm still believe it isn't the best idea. It might cause the policy to do not be properly updated later. As the policy will be in the right revision, fleet-server will consider there is nothing to be done, whereas there is.

It'll keep the data more consistent if we have it back where it was

Reply:
Just updated